components/resizable_sidebar/resizable_lhs/index.tsx

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {HTMLAttributes, useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {useSelector} from 'react-redux';
+
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/common';
+
+import {getLhsSize} from 'selectors/lhs';
+import LocalStorageStore from 'stores/local_storage_store';
+import {DEFAULT_LHS_WIDTH, LHS_MIN_MAX_WIDTH} from 'utils/constants';
+import {isOverLimit, isResizableSize, requestAnimationFrameForMouseMove, shouldSnapWhenSizeGrown, shouldSnapWhenSizeShrunk} from '../utils';
+
+interface Props extends HTMLAttributes<'div'> {
+    children: React.ReactNode;
+}
+
+function Resizable({
+    children,
+    id,
+    className,
+}: Props) {
+    const lhsRef = useRef<HTMLDivElement>(null);
+    const resizeLineRef = useRef<HTMLDivElement>(null);
+
+    const [isResizeLineSelected, setIsResizeLineSelected] = useState(false);
+    const previousClientX = useRef(0);
+
+    const lhsSize = useSelector(getLhsSize);
+    const userId = useSelector(getCurrentUserId);
+
+    const minWidth = useMemo(() => LHS_MIN_MAX_WIDTH[lhsSize].min, [lhsSize]);
+    const maxWidth = useMemo(() => LHS_MIN_MAX_WIDTH[lhsSize].max, [lhsSize]);
+
+    const isLhsResizable = useMemo(() => isResizableSize(lhsSize), [lhsSize]);
+
+    const handleDoubleClick = useCallback(() => {
+        if (lhsRef.current) {
+            lhsRef.current.style.width = `${DEFAULT_LHS_WIDTH}px`;
+        }
+
+        LocalStorageStore.setLhsWidth(userId, DEFAULT_LHS_WIDTH);
+    }, [userId]);
+
+    const handleMouseDown = useCallback((e: React.MouseEvent) => {
+        if (!isLhsResizable) {
+            return;
+        }
+
+        previousClientX.current = e.clientX;
+        setIsResizeLineSelected(true);
+        document.body.style.cursor = 'col-resize';
+
+        if (lhsRef.current) {
+            lhsRef.current.classList.add('sidebar--left-dragged');
```

Comment:
I have the same question here regarding setting this class using a ref. It seems more idiomatic to pass this below using  the `className` prop instead of setting it via a ref here

Reply:
I changed it to use `className` unless using external state is available.